Assignment6.Rmd

```diff
@@ -78,8 +107,72 @@ Choose a data file from the (University of Michigan Open Data Set)[https://githu
 
 ```{r}
 
-```
+student.record <- read.csv("~/tmp/assignment6/student.record.csv", header=TRUE, stringsAsFactors = TRUE)
+
+
+
+#This is my raw variable analysis
+
+s.record<-student.record%>%
+  gather(key="Student ID", value = "Major", MAJOR1_DEPT, MAJOR2_DEPT, MAJOR3_DEPT)%>%
+  select(ANONID, LAST_ACT_ENGL_SCORE,LAST_ACT_MATH_SCORE,LAST_ACT_READ_SCORE,LAST_ACT_SCIRE_SCORE,LAST_ACT_COMP_SCORE,
+         HSGPA, STDNT_GROUP1)
+
+
+s.r1<-na.omit(s.record)
+colnames(s.r1)<-c("ID", "English", "Math", "Reading", "Science", "Comprehensive", "GPA", "Group")
+
+c.tree3 <- rpart(Group ~ GPA+Comprehensive+Science+Reading+English+Math+ID, method="class", data=s.r1)
+printcp(c.tree3)
 
+rpart.plot(c.tree3)
+
+#This is my PCA
+s.r2<-s.r1%>%
+  select(-c(Group, ID))
+
+pca <-prcomp(s.r2, scale. = TRUE)
+
+summary(pca)
+
+pca$rotation
+
+c.tree4 <- rpart(Group ~ English+Comprehensive+Science, method="class", data=s.r1)
+printcp(c.tree4)
+
+rpart.plot(c.tree4)
+
+s.r1$predict1 <- predict(c.tree3, s.r1, type = "class")
+s.r1$predict2 <- predict(c.tree4, s.r1, type = "class")
+
+for (i in 1:nrow(s.r2)){
+  if(s.r1$predict1[i]==s.r1$Group[i]){
+    s.r1$matchx[i]<-"Y"
+  }else{
+    s.r1$matchx[i]<-"N"
+  }
+}
+
+for (i in 1:nrow(s.r1)){
+  if(s.r1$predict2[i]==s.r1$Group[i]){
+    s.r1$matchy[i]<-"Y"
+  }else{
+    s.r1$matchy[i]<-"N"
+  }
+}
+
+# x is for predict1, y is for predict2
+
+s.r1x<-s.r1 %>%
+  group_by(matchx)%>%
+  count(matchx)
+
+s.r1y<-s.r1 %>%
+  group_by(matchy)%>%
+  count(matchy)
+
+```
+Of the two models, technically, the first raw analysis is correct more often than the second model. However, I have to question how valuable that analysis is for a couple of reasons. First and foremost, the margin of correctness that it has over the feature selected model is very small. Second, both decision trees are tested against the training sample, rather than a new test sample. I have to assume that there is a chance that decision tree is overfitted, and would want a new sample to see whether this second decision tree is actually weaker.
```

Comment:
Hi Leo, I think you did a good job and generally have a good understanding of model comparison of classification trees by looking at model accuracy and error rate. You are right in the sense that the two models you tried here do not differ much regarding performance of precision. When doing the research, you would have more specific questions in mind that help you justify and selecting features that you are interested in and also compare the results based on your research question. Besides, precision is not the only thing you want to look at when deciding which model you want to go with and you could think about finding other measures that could help you make the decision (such as model parsimony...)
In addition, using PCA for feature selection is a good idea (but it seems you did not use it to help you improve the model... could try to use the PCs as predictors and see whether the precision improves)